### ISSUE TITLE
Caused by: java.lang.IllegalArgumentException: Cannot access script json2xml.py

### ISSUE THREAD

**Issue by ashley-stendel** [spring-cloud-stream-app-starters/python#12]
Hi!

We are trying to pass in a python script with 1.0.1.RELEASE. The file is executable. 
we created the following stream 
`stream create --name converter --definition " http | python-jython --git.uri=https://github.com/ritceynoah/json2xml --jython.script=json2xml.py | log"`

and 

`stream create --name converter --definition " http | python-jython --jython.script=~/Documents/json2xml.py | log"`

We are deploying to kubernetes and we get the following error in our logs.
`Caused by: java.lang.IllegalArgumentException: Cannot access script json2xml.py`

I think it can't find the file.... any suggestions on where we went wrong?

Thank you!

**Comment by sabbyanandan** [comment]
Hi, @ashley-stendel. 

Do you see the error in both the stream definitions? 

For the 2nd stream, it will fail for sure because the app will expect the file to be in the defined directory at the pod/container level. For it to work, the script _should_ be remotely accessible, which is what 1st definition is doing, but is it not working?

Also wondering if between http and https URIs will make a difference. Something to try, though.

**Comment by ashley-stendel** [comment]
Hi @sabbyanandan 
Thank you for your quick reply!

Yes, we see the same error in both stream definitions. We just tried with http instead of https same error.... ie: `stream create --name converter --definition "http | python-jython --git-uri=http://github.com/ritceynoah/json2xml --jython.script=json2xml.py | log"`

Any other suggestions?
Thank you very much!

**Comment by sabbyanandan** [comment]
Thanks. A quick review of how [`PathMatchingResourcePatternResolver`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/io/support/PathMatchingResourcePatternResolver.html) resolves artifacts (we use it [here](https://github.com/spring-cloud-stream-app-starters/python/blob/master/python-app-starters-common/src/main/java/org/springframework/cloud/stream/app/python/script/ScriptProperties.java#L66)), it appears the script may have to be a fully qualified path. 

You could enable DEBUG logs for --logging.org.springframework.core.io.support=DEBUG` - you can supply it as a property for `python-jython` in the stream definition.

Also, it might be easier to troubleshoot it in the Local server - quicker feedback. If it works Local, it will on K8S, too.

@dturanski: Any other ideas?

**Comment by artembilan** [comment]
We have code there like this:
```
return new PathMatchingResourcePatternResolver().getResource("file:" + resourceName);
```

So, your `--jython.script=~/Documents/json2xml.py` will become as a resource link like: `file:~/Documents/json2xml.py`.

On the other hand you say:
> We are deploying to kubernetes

Isn't that about a cloud? I assume the local filesystem there in the pod is different than yours local one.

I might be wrong, just sharing my observation and feeling.

Sorry for the noise otherwise.

**Comment by sabbyanandan** [comment]
Hi, @ashley-stendel. The fix in spring-cloud-stream-app-starters/python#13 may address the problem specifically in K8S. Could you try the PR and share your findings?

**Comment by sabbyanandan** [comment]
Update: With the recent fixes (via `docker:springcloudstream/python-jython-processor-rabbit:latest`; see [here](https://hub.docker.com/r/springcloudstream/python-jython-processor-rabbit/tags/)), I'm able to run the following stream successfully on minikube.

**apps:**
```
dataflow:>app list --id processor:python-jython
╔══════╤═══════════════════════════════╤════╤════╗
║source│           processor           │sink│task║
╠══════╪═══════════════════════════════╪════╪════╣
║      │> python-jython-1.0.1.RELEASE <│    │    ║
║      │python-jython-latest           │    │    ║
╚══════╧═══════════════════════════════╧════╧════╝
```

**stream:**
> dataflow:>stream create --name foo --definition "http | python-jython --git.uri=https://github.com/sabbyanandan/python --jython.script=upper.py | log"

**data:**
>  curl -X POST "http://192.168.99.100:32123" -d "hello world" -H "Content-Type:text/plain"

**logs:**
```
2018-03-14 15:16:54.269  INFO 1 --- [on-jython.foo-1] log-sink                                 : HELLO WORLD
```

**Comment by sabbyanandan** [comment]
@ashley-stendel: One last update. The fixes were released in the 1.0.2.RELEASE and it is on its way to Maven Central. However, the Docker [image-tag](https://hub.docker.com/r/springcloudstream/python-jython-processor-rabbit/tags/) is already available for use. Let us know if you have any questions. 

Thanks for trying the Python processor and the feedback.
